Upgrade docker image

sf1919 opened the pull request:
Changes to the docker images as follows
- upgrade to java 11
- remove pre-commit
- Add a label

martyngigg commented on jenkins-node/mantid-builder/docker-images/CentOS7Slim.Dockerfile:
```diff
@@ -1,11 +1,11 @@
-ARG DEVELOPMENT_IMAGE_VERSION=0.9
+ARG DEVELOPMENT_IMAGE_VERSION=0.10
 FROM ghcr.io/mantidproject/mantid-development-centos7-slim:${DEVELOPMENT_IMAGE_VERSION}
 
```
Can we add the appropriate `LABEL` command to this image too?

martyngigg reviewed (changes requested):
This looks good. Just one small request.

martyngigg reviewed (changes requested):
I've just come across one more thing that would be good to include now.

Apparently our git version is ancient and breaks newer pre-commit (https://github.com/pre-commit/pre-commit-hooks/issues/718).
Can we install a more up-to-date git?

sf1919 commented:
I've updated git, added openssl and redone the labels so they work as intended (see https://github.com/mantidproject/dockerfiles/pkgs/container/jenkins-node-centos7-slim for example).

martyngigg commented on development/docker/CentOS7Slim.Dockerfile:
```diff
@@ -10,24 +10,34 @@ FROM neszt/cppcheck-docker:${CPPCHECK_VERSION} AS upstream_cppcheck
 RUN useradd --uid 911 --user-group --shell /bin/bash --create-home abc
 
+#Add label for transparency
+LABEL org.opencontainers.image.source https://github.com/mantidproject/dockerfiles
```
Could we move this nearer the top, just under the FROM, as it's metatdata and it would be good to distinguish it from script-like commands.

martyngigg reviewed (changes requested):
It's great we've got this working now.

Just a few minor tweaks.

sf1919 commented on development/docker/CentOS7Slim.Dockerfile:
```diff
@@ -10,24 +10,34 @@ FROM neszt/cppcheck-docker:${CPPCHECK_VERSION} AS upstream_cppcheck
 FROM centos:7
 
+# Install minimal developer tools
+RUN yum install -y \
```
Unfortunately the script is very sensitive as to where I put the remove git command. If it's not directly after the installation of the `isu-release-el7.rpm` I get an error that it does not recognise the command `git`

martyngigg replied:
Do we need to remove `git`? I tried starting a plain `centos:7` container and it says `git` is not installed:
```
$ docker run -it centos:7 git --version
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "git": executable file not found in $PATH: unknown.
```

I had a quick go locally and found that having

```
# Install minimal developer tools
RUN yum install -y \
  https://repo.ius.io/ius-release-el7.rpm  \
  https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
RUN yum install -y \
  ccache \
  curl \
  git236 \
  graphviz \
  libXScrnSaver \
  openssl \
  pciutils-libs \
  perl-Digest-MD5 \
  sudo \
  which \
  xorg-x11-server-Xvfb && \
  # Clean up
  rm -rf /tmp/* /var/tmp/*
```

gave me a container than when I ran:

```sh
$ docker run -it ghcr.io/mantidproject/mantid-development-centos7-slim:latest git --version
+ TARGET_USERNAME=abc
+ for rule in '/etc/entrypoint.d/*.sh'
+ source /etc/entrypoint.d/000_change_user_ids.sh
++ set -x
++ PUID=911
++ PGID=911
++ groupmod --non-unique --gid 911 abc
++ usermod --non-unique --uid 911 abc
usermod: no changes
+ for rule in '/etc/entrypoint.d/*.sh'
+ source /etc/entrypoint.d/010_abc_own_directories.sh
++ set -x
++ chown abc:abc /mantid_src
++ chown abc:abc /mantid_build
++ chown abc:abc /mantid_data
++ chown abc:abc /ccache
+ for rule in '/etc/entrypoint.d/*.sh'
+ source /etc/entrypoint.d/020_locale.sh
++ set -x
++ export LANG=C.UTF-8
++ LANG=C.UTF-8
+ CMD='git --version'
+ runuser -u abc -- git --version
git version 2.36.1
```

sf1919 replied:
I agree, it seems we don't need to remove git as it doesn't get auto installed by anything else by accident. I've even moved the install of git236 into the main yum install layer to make it neater.

martyngigg commented on development/docker/CentOS7Slim.Dockerfile:
```diff
@@ -10,24 +10,34 @@ FROM neszt/cppcheck-docker:${CPPCHECK_VERSION} AS upstream_cppcheck
+  openssl \
   pciutils-libs \
   perl-Digest-MD5 \
   python36-pip \
```
Just spotted this one. Can we remove this one? We no longer need any python36 stuff.

martyngigg reviewed (approved):
This looks fab now 👍🏻
